### ISSUE TITLE
Bug: 

### ISSUE THREAD

**Issue by jbromley** [hoob3rt/lualine.nvim#263]
### Self Checks

- [ X ] I'm using the latest lualine and neovim nightly.
- [ X ] I didn't find the issue in existing issues or prs.

### How to reproduce the problem
Run neovim with the configuration reproduced below, which attempts to use the `auto` theme.

### Actual behaviour
Neovim launches and produces the error message `theme auto not found, defaulting to gruvbox`. The gruvbox theme is applied.

### Expected behaviour

lualine loads lualine/themes/auto.lua and attempts to use the current theme colors.

### Aditional information

The file `auto.lua` exists in the themes directory and loading several other themes has worked without any problems. It appears the line that requires the theme file is failing.

### Minimal config to reproduce the issue

<details>
<summary>Config</summary>

```
-- Set up the package manager and install lualine.
vim.cmd 'packadd paq-nvim'
local paq = require('paq-nvim').paq
paq {'savq/paq-nvim', opt = true}
paq {'kyazdani42/nvim-web-devicons', opt = true}
paq {'hoob3rt/lualine.nvim'}

require('lualine').setup {options = { theme = 'auto' }}
```
</details>

**Comment by jbromley** [comment]
I looked into this a little further and this is what I've found. 

The problem is that the call to `utils.extract_highlight_colors('Normal', 'bg')` at `lualina/themes/auto.lua:120` was returning `nil` and calling `get_color_brightness` (which calls `rgb_str2num`) on this nil was causing the load of `auto.lua` to fail. One odd thing is that I inspected the value returned for the Normal highlight group by `vim.api.nvim_get_hl_by_name` and got the following:
```
{ [true] = 6 }
```
so I don't know if there is a deeper problem with the vim.api call. I could get `auto.lua` to load by checking if the returned highlight group color was nil before calling `get_color_brightness`, like so
```
-- Change brightness of colors
-- darken in case of light theme, lighten in case of dark theme
local normal_color = utils.extract_highlight_colors('Normal', 'bg')
if normal_color ~= nil and get_color_brightness(normal_color) > 0.5 then
  brightness_modifier_parameter = -brightness_modifier_parameter
end
```

**Comment by shadmansaleh** [comment]
Do you have a trasparant background in your colorscheme ?

**Comment by jbromley** [comment]
Taking a tip from your question, I did some more investigation. I was not setting any neovim colorscheme when I saw this problem. If a set the colorscheme to anything other than default, I do not see this problem. If I don't explicitly set a colorscheme or set it explicitly to default, I see the problem. So to answer your question, yes, I guess the default colorscheme does have transparency in it.

That said, I think a check like I suggested above is still a good idea. That way, when you're running your terminal with a default theme and you run neovim with the default colorscheme and lualine with the auto theme, you don't get error messages about not being able to load the auto theme.

**Comment by shadmansaleh** [comment]
> That said, I think a check like I suggested above is still a good idea

I agree . When I was writting that I wasn't aware of transparant backgrounds . But they area  thing in neovim . I don't think we can reliably grab terminal bg when transparent background so a check is probably best thing to do . Open a pr with your changes I think @hoob3rt will agree :)
